include/aspect/postprocess/particle_PDF.h

```diff
@@ -0,0 +1,178 @@
+          const unsigned int x_index,
+          const unsigned int y_index,
+          const unsigned int z_index) const
+        {
+          TableIndices<dim> entry_index;
+          entry_index[0] = x_index;
+          entry_index[1] = y_index;
+          if (dim == 3)
+            entry_index[2] = z_index;
+
+          return function_output_table(entry_index);
+        }
+
+
+
+        /**
+        * Calculates the relevant statistical values from the contents of the PDF.
+        */
+        void set_statistical_values()
+        {
+          max = std::numeric_limits<double>::min();
+          min = std::numeric_limits<double>::max();
+          standard_deviation = 0;
+          mean = 0;
+          median = 0;
+          quartile_first = 0;
+          quartile_second = 0;
+          quartile_IQR = 0;
+
+          // Loop through all values of the function to set initial stats
+          for (unsigned int x = 0; x< pdf_granularity; ++x)
+            {
+              for (unsigned int y = 0; y< pdf_granularity; ++y)
+                {
+                  if (dim == 3)
+                    {
+                      for (unsigned int z = 0; z< pdf_granularity; ++z)
+                        {
+                          double this_value = evaluate_function_at_index(x,y,z);
+                          max = std::max(max, this_value);
+                          min = std::min(min, this_value);
+                          // Sum in mean, then divide after this loop
+                          mean += this_value;
+                        }
+                    }
+                  else
+                    {
+                      double this_value = evaluate_function_at_index(x,y,0);
+                      max = std::max(max, this_value);
+                      min = std::min(min, this_value);
+                      // Sum in mean, then divide after this loop
+                      mean += this_value;
+                    }
+                }
+            }
+          // Set the true mean
+          mean /= (Utilities::fixed_power<dim>(pdf_granularity));
+          double squared_deviation_sum = 0;
+
+          // This sum all the squared deviations for standard deviation
+          for (unsigned int x = 0; x< pdf_granularity; ++x)
+            {
+              for (unsigned int y = 0; y< pdf_granularity; ++y)
+                {
+                  TableIndices<dim> entry_index;
+                  entry_index[0] = x;
+                  entry_index[1] = y;
+                  double this_value = function_output_table(entry_index);
+                  double deviation_squared = (this_value-mean)*(this_value-mean);
+                  squared_deviation_sum += deviation_squared;
+                }
+            }
+          // Standard deviation of all the defined points in the density function
+          squared_deviation_sum /= (pdf_granularity*dim);
+          standard_deviation = std::sqrt(squared_deviation_sum);
+        };
```

Comment:
Not wrong, but the semicolon is also not necessary:
```suggestion
        }
```